src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -600,66 +558,35 @@ protected override SupervisorStrategy SupervisorStrategy()
-
-                        directive = Directive.Stop;
-                    })
-                    .With<ShutDownAssociation>(shutdown =>
-                    {
+                        return Directive.Stop;
+                    case ShutDownAssociation shutdown:
                         KeepQuarantinedOr(shutdown.RemoteAddress, () =>
                         {
                             _log.Debug("Remote system with address [{0}] has shut down. Address is now gated for {1}ms, all messages to this address will be delivered to dead letters.",
-                                 shutdown.RemoteAddress, _settings.RetryGateClosedFor.TotalMilliseconds);
+                                shutdown.RemoteAddress, _settings.RetryGateClosedFor.TotalMilliseconds);
                             _endpoints.MarkAsFailed(Sender, Deadline.Now + _settings.RetryGateClosedFor);
                         });
-                        directive = Directive.Stop;
-                    })
-                    .With<HopelessAssociation>(hopeless =>
-                    {
-                        if (hopeless.Uid.HasValue)
-                        {
-                            _log.Error(hopeless.InnerException, "Association to [{0}] with UID [{1}] is irrecoverably failed. Quarantining address.",
-                                hopeless.RemoteAddress, hopeless.Uid);
-                            if (_settings.QuarantineDuration.HasValue)
-                            {
-                                _endpoints.MarkAsQuarantined(hopeless.RemoteAddress, hopeless.Uid.Value,
-                               Deadline.Now + _settings.QuarantineDuration.Value);
-                                _eventPublisher.NotifyListeners(new QuarantinedEvent(hopeless.RemoteAddress,
-                                    hopeless.Uid.Value));
-                            }
-                        }
-                        else
-                        {
-                            _log.Warning("Association to [{0}] with unknown UID is irrecoverably failed. Address cannot be quarantined without knowing the UID, gating instead for {1} ms.",
-                                hopeless.RemoteAddress, _settings.RetryGateClosedFor.TotalMilliseconds);
-                            _endpoints.MarkAsFailed(Sender, Deadline.Now + _settings.RetryGateClosedFor);
-                        }
-                        directive = Directive.Stop;
-                    })
-                    .Default(msg =>
-                    {
-                        if (msg is EndpointDisassociatedException || msg is EndpointAssociationException) { } //no logging
+                        return Directive.Stop;
+                    case HopelessAssociation h:
+                        return Hopeless(h);
+                    case ActorInitializationException i when i.InnerException is HopelessAssociation h2:
```

Comment:
New case for handling `ReliableDeliverySupervisor` failing quarantine check in its constructor.